File: 12. Integer to Roman/solution.c
char to_numeral(int num)
{
    switch (num)
    {
        case 1: return "I";
        case 5: return "V";
        case 10: return "X";
        case 50: return "L";
        case 100: return "C";
        case 500: return "D";
        case 1000: return "M";
    }

    return "";
}


char* intToRoman(int num)
{
    int len = 0;

}

File: 12. Integer to Roman/solution.py
class Solution:
    ROMAN_VALUES: list[tuple[int, str]] = [
        (1000, "M"),
        (900, "CM"),
        (500, "D"),
        (400, "CD"),
        (100, "C"),
        (90, "XC"),
        (50, "L"),
        (40, "XL"),
        (10, "X"),
        (9, "IX"),
        (5, "V"),
        (4, "IV"),
        (1, "I")
    ]

    def intToRoman(self, num: int, memo: dict[int, str] = {}) -> str:
        if num in memo.keys():
            return memo[num]

        res = []
        for value, numeral in self.ROMAN_VALUES:
            while num >= value:
                res.append(numeral)
                num -= value

        memo[num] = "".join(res)
        return memo[num]
